zookeeper/source/zk_nn_client.cc

```diff
@@ -401,15 +403,17 @@ namespace zkclient{
      * if necessary.
      */
     bool ZkNnClient::mkdir_helper(const std::string& path, bool create_parent) {
-        if (create_parent) {
+       	LOG(INFO) << "mkdir_helper called with input " << path; 
+		if (create_parent) {
             std::vector<std::string> split_path;
             boost::split(split_path, path, boost::is_any_of("/"));
             bool not_exist = false;
             std::string unroll;
             std::string p_path = "";
-            for (int i = 0; i < split_path.size(); i++) {
-                p_path += split_path[i] + "/";
-                if (!file_exists(p_path)) {
+            for (int i = 1; i < split_path.size(); i++) {
+                p_path += "/" + split_path[i];
```

Comment:
Should comment that we start at index 1 because it includes "/" as the first element in the array when we do NOT want that